rotors_hil_interface/include/rotors_hil_interface/hil_listeners.h

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2016 Pavel Vechersky, ASL, ETH Zurich, Switzerland
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ROTORS_HIL_LISTENERS_H_
+#define ROTORS_HIL_LISTENERS_H_
+
+#include <boost/thread/mutex.hpp>
+#include <Eigen/Dense>
+#include <geometry_msgs/TwistStamped.h>
+#include <ros/ros.h>
+#include <sensor_msgs/FluidPressure.h>
+#include <sensor_msgs/Imu.h>
+#include <sensor_msgs/MagneticField.h>
+#include <sensor_msgs/NavSatFix.h>
+
+namespace rotors_hil {
+// Constants
+static constexpr float kAirDensity = 1.18;
+static constexpr float kGravityMagnitude = 9.8068;
+static constexpr float kStandardPressureMBar = 1013.25;
+static constexpr float kTemperature = 15.0;
+static constexpr int kFixNone = 0;
+static constexpr int kFix3D = 3;
+static constexpr int kHDOP = 100;
+static constexpr int kVDOP = 100;
+static constexpr int kSatellitesVisible = 4;
+static constexpr int kUnknown = 65535;
+
+// Conversions
+static constexpr float kDegreesToHil = 1e7;
+static constexpr float kFeetToMeters = 0.3048;
+static constexpr float kMetersToCm = 100.0;
+static constexpr float kMetersToMm = 1000.0;
+static constexpr float kPascalToMillibar = 0.01;
+static constexpr float kPressureToAltExp = 0.190284;
+static constexpr float kPressureToAltMult = 145366.45;
+static constexpr float kSecToNsec = 1e9;
+static constexpr float kTeslaToGauss = 10000.0;
+
+struct HilData {
+  HilData() :
+      temperature(kTemperature),
+      eph(kHDOP),
+      epv(kVDOP),
+      cog(kUnknown),
+      satellites_visible(kSatellitesVisible) {}
+
+  Eigen::Quaterniond att;      // Attitude quaternion
+  Eigen::Vector3f acc;         // Linear acceleration (m/s^2)
```

Comment:
In the MAVLink message spec (https://pixhawk.ethz.ch/mavlink/#HIL_SENSOR) most of these fields are floats so I figured I'd store them as floats here as well. Or should I change this to double on my side?